tests/by-util/test_groups.rs

```diff
@@ -1,56 +1,174 @@
+// ubuntu image for github action available with a higher version than 8.30.
+const VERSION_EXPECTED: &str = "8.30"; // Version expected for the reference `groups` in $PATH
+const VERSION_MULTIPLE_USERS: &str = "8.31";
+const UUTILS_WARNING: &str = "uutils-tests-warning";
+const UUTILS_INFO: &str = "uutils-tests-info";
+
+macro_rules! unwrap_or_return {
+    ( $e:expr ) => {
+        match $e {
+            Ok(x) => x,
+            Err(e) => {
+                println!("{}: test skipped: {}", UUTILS_INFO, e);
+                return;
+            }
+        }
+    };
+}
+
+fn whoami() -> String {
+    // Apparently some CI environments have configuration issues, e.g. with 'whoami' and 'id'.
+    //
+    // From the Logs: "Build (ubuntu-18.04, x86_64-unknown-linux-gnu, feat_os_unix, use-cross)"
+    //    whoami: cannot find name for user ID 1001
+    // id --name: cannot find name for user ID 1001
+    // id --name: cannot find name for group ID 116
+    //
+    // However, when running "id" from within "/bin/bash" it looks fine:
+    // id: "uid=1001(runner) gid=118(docker) groups=118(docker),4(adm),101(systemd-journal)"
+    // whoami: "runner"
+
+    // Use environment variable to get current user instead of
+    // invoking `whoami` and fall back to user "nobody" on error.
+    std::env::var("USER").unwrap_or_else(|e| {
+        println!("{}: {}, using \"nobody\" instead", UUTILS_WARNING, e);
+        "nobody".to_string()
+    })
+}
+
 #[test]
 #[cfg(unix)]
 fn test_groups() {
-    if !is_ci() {
-        new_ucmd!().succeeds().stdout_is(expected_result(&[]));
-    } else {
-        // TODO: investigate how this could be tested in CI
-        // stderr = groups: cannot find name for group ID 116
-        println!("test skipped:");
-    }
+    let result = new_ucmd!().run();
+    let exp_result = unwrap_or_return!(expected_result(&[]));
+
+    result
+        .stdout_is(exp_result.stdout_str())
+        .stderr_is(exp_result.stderr_str())
+        .code_is(exp_result.code());
 }
 
 #[test]
 #[cfg(unix)]
-#[ignore = "fixme: 'groups USERNAME' needs more debugging"]
 fn test_groups_username() {
-    let scene = TestScenario::new(util_name!());
-    let whoami_result = scene.cmd("whoami").run();
+    let test_users = [&whoami()[..]];
 
-    let username = if whoami_result.succeeded() {
-        whoami_result.stdout_move_str()
-    } else if is_ci() {
-        String::from("docker")
-    } else {
-        println!("test skipped:");
+    let result = new_ucmd!().args(&test_users).run();
+    let exp_result = unwrap_or_return!(expected_result(&test_users));
+
+    result
+        .stdout_is(exp_result.stdout_str())
+        .stderr_is(exp_result.stderr_str())
+        .code_is(exp_result.code());
+}
+
+#[test]
+#[cfg(unix)]
+fn test_groups_username_multiple() {
+    #[cfg(target_os = "linux")]
+    let util_name = util_name!();
+    #[cfg(all(unix, not(target_os = "linux")))]
+    let util_name = &format!("g{}", util_name!());
```

Comment:
Is there some reason not to just use the utility names instead of `util_name!()`?

Reply:
see other comment

Reply:
Just use the simple strings instead of constructing them from `format!()` and `util_name!()`.

Reply:
Add the ToDO here as well please.